examples/mugen/retrieval/eval.py

```diff
@@ -0,0 +1,120 @@
+
+from examples.mugen.data.mugen_datamodules import MUGENDataModule
+from examples.mugen.data.mugen_dataset import MUGENDatasetArgs
+
+from examples.mugen.retrieval.model import VideoCLIPLightningModule
+from examples.mugen.retrieval.video_clip import PRETRAINED_S3D_KINETICS400_URL
+from pytorch_lightning import Trainer
+
+from torchmultimodal.transforms.bert_text_transform import BertTextTransform
+from torchmultimodal.transforms.video_transform import VideoTransform
+
+
+arg_structure = {
+    "datamodule": {
+        "batch_size": {"type": int, "default": 16},
+        "num_workers": {"type": int, "default": 4},
+        "shuffle": {"action": "store_true"},
+    },
+    "lightningmodule": {
+        "logit_scale": {"type": float, "default": 0.07},
+        "logit_scale_max": {"type": float, "default": 100.0},
+    },
+    "dataset": {
+        "data_path": {
+            "type": str,
+            "default": "datasets/coinrun/coinrun_dataset_jsons/release",
+        },
+        "sample_every_n_frames": {"type": int, "default": 3},
+        "sequence_length": {"type": int, "default": 32},
+        "resolution": {"type": int, "default": 224},
+        "bbox_smap_for_agent": {"action": "store_true"},
+        "bbox_smap_for_monsters": {"action": "store_true"},
+        "use_manual_annotation": {"action": "store_true"},
+        "use_auto_annotation": {"action": "store_true"},
+        "use_downsampled_trainset": {"action": "store_true"},
+        "fixed_start_idx": {"action": "store_true"},
+        "get_game_frame": {"action": "store_true"},
+        "get_seg_map": {"action": "store_true"},
+        "get_text_desc": {"action": "store_true"},
+        "debug": {"action": "store_true"},
+    },
+    "videoclip": {
+        "text_pretrained": {"action": "store_true"},
+        "text_trainable": {"action": "store_true"},
+        "text_model_name": {"type": str, "default": "distilbert-base-uncased"},
+        "text_model_config": {"type": json.loads, "default": None},
+        "text_padding_value": {"type": int, "default": 0},
+        "video_pretrained": {"action": "store_true"},
+        "video_trainable": {"action": "store_true"},
+        "video_pretrain_path": {"type": str, "default": PRETRAINED_S3D_KINETICS400_URL},
+        "proj_out_dim": {"type": int, "default": 256},
+        "proj_dropout": {"type": float, "default": 0.1},
+    },
+    "evaluation": {
+        "accelerator": {"type": str, "default": "auto"},
+        "devices": {"type": int, "default": 1},
+        "checkpoint_path": {
+            "type": str,
+            "default": "lightning_videoclip_mugen_ckpt.pt",
+        },
+    },
+}
+
+
+def parse_all_args():
+    parser = argparse.ArgumentParser()
+    for arguments_dict in arg_structure.values():
+        for arg_name, arg_options in arguments_dict.items():
+            parser.add_argument("--" + arg_name, **arg_options)
+    args = parser.parse_args()
+    return args
+
+
+def get_args_from_group(args, group_name):
```

Comment:
Agree with @langong347 on moving to YAML if possible, this method is probably unnecessary. You can also take a look at `vqa.yaml` and `finetune.py` in #224 as an example of how to make this change.